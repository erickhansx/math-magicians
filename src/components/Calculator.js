import React, { useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.scss';

const Calculator = () => {
  const [InitialValue, newValue] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const onClickEvent = (e) => {
    const valueBtn = e.target.innerHTML;
    const result = calculate(InitialValue, valueBtn);
    newValue(result);
  };
  const { total, operation, next } = InitialValue;
  return (
    <div className="calculator">
      <div className="calculator__header">
        <h2>Let&apos;s do some Math!</h2>
      </div>
      <div className="calculator__body">
        <div className="input-area">
          <p className="input-calc" data-testid="testp">
            {total}
            {operation}
            {next}
          </p>
        </div>
        <div className="grey-btns">
          <button onClick={onClickEvent} type="button" className="btn">
            AC
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            +/-
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            %
          </button>
          <button onClick={onClickEvent} type="button" className="btn operator">
            รท
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            7
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            8
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            9
          </button>
          <button onClick={onClickEvent} type="button" className="btn operator">
            x
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            4
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            5
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            6
          </button>
          <button onClick={onClickEvent} type="button" className="btn operator">
            -
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            1
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            2
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            3
          </button>
          <button onClick={onClickEvent} type="button" className="btn operator">
            +
          </button>
          <button onClick={onClickEvent} type="button" className="btn long">
            0
          </button>
          <button onClick={onClickEvent} type="button" className="btn">
            .
          </button>
          <button onClick={onClickEvent} type="button" className="btn operator">
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
