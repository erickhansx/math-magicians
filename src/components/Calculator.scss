.App {
  display: flex;
  justify-content: center;
  height: 80vh;
}

.calculator {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

::placeholder {
  color: #fff;
  font-size: 20px;
}

.input-area {
  input {
    width: 98%;
    height: 90%;
    margin-bottom: 25px;
    direction: rtl;
    background-color: grey;
  }
}

.btn {
  width: 25%;
  height: 45%;
  border: 1px solid rgb(217, 217, 217);
}

.operator {
  background-color: #f5913e;
  border: 1px solid rgb(217, 217, 217);
}

.long {
  width: 50%;
}
