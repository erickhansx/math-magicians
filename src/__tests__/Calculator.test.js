import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Quote Renders', () => {
  const CalculatorRender = renderer.create(<Calculator />).toJSON();
  expect(CalculatorRender).toMatchSnapshot();
});
