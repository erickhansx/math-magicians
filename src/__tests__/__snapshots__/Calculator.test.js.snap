// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Renders 1`] = `
<div
  className="calculator"
>
  <div
    className="calculator__header"
  >
    <h2>
      Let's do some Math!
    </h2>
  </div>
  <div
    className="calculator__body"
  >
    <div
      className="input-area"
    >
      <p
        className="input-calc"
        data-testid="testp"
      >
        0
      </p>
    </div>
    <div
      className="grey-btns"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn long"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
